src/project/manifest/pyproject.rs

```diff
@@ -0,0 +1,167 @@
+use pep508_rs::{Requirement, VersionOrUrl};
+use rattler_conda_types::{NamelessMatchSpec, PackageName, ParseStrictness::Lenient, VersionSpec};
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::str::FromStr;
+use tokio::runtime::Handle;
+use toml_edit;
+use toml_edit::TomlError;
+
+use crate::pypi_name_mapping::conda_pypi_name_mapping;
+
+use super::{error::RequirementConversionError, ProjectManifest, SpecType};
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct PyProjectManifest {
+    #[serde(flatten)]
+    inner: pyproject_toml::PyProjectToml,
+    tool: Tool,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+#[serde(rename_all = "kebab-case")]
+struct Tool {
+    pixi: ProjectManifest,
+}
+
+impl std::ops::Deref for PyProjectManifest {
+    type Target = pyproject_toml::PyProjectToml;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+impl PyProjectManifest {
+    /// Parses a toml string into a pyproject manifest.
+    pub fn from_toml_str(source: &str) -> Result<Self, TomlError> {
+        toml_edit::de::from_str(source).map_err(TomlError::from)
+    }
+}
+
+impl From<PyProjectManifest> for ProjectManifest {
+    fn from(item: PyProjectManifest) -> Self {
+        // Start by loading the data nested under "tool.pixi"
+        let mut manifest = item.tool.pixi.clone();
+
+        // TODO: tool.pixi.project.name should be made optional or read from project.name
+        // TODO: could copy across / convert some other optional fields if relevant
+
+        // Gather pyproject dependencies
+        let mut requirements = item
+            .project
+            .as_ref()
+            .and_then(|p| p.dependencies.as_ref())
+            .cloned()
+            .unwrap_or_else(Vec::new);
+
+        // Add python as dependency based on the project.requires_python property (if any)
+        let pythonspec = item
+            .project
+            .as_ref()
+            .and_then(|p| p.requires_python.as_ref())
+            .map(|v| VersionOrUrl::VersionSpecifier(v.clone()));
+        let python_req = Requirement {
```

Comment:
From a pixi perspective it would make sense to add this to a `[tool.pixi.depenencies]` and save the version there aswell. A first thought would be to add this piece of code to the `pixi init` which could auto add the following to the `pyproject.toml`:
```toml
# This project supports installation with pixi, more info at https://pixi.sh
[tool.pixi.project]
channels = ["conda-forge"]
platforms = ["CURRENT"]

[tool.pixi.dependencies]
python = "x.y.z"
```

Reply:
My thinking was to avoid duplicating information between the pyproject data (e.g. `requires-python`, `dependencies`, etc.) and the pixi data (e.g. `dependencies`), in the cases we successfully translate pep 508 requirement into a conda MatchSpec.

I think in many cases this is doable automatically; noting that we would need an escape hatch for the cases that cannot be handled automatically.

This is the approach that conda-lock uses, and it has worked really well for me in the past.

Duplicating dependencies would mean that users have two places to modify to update a dependency, and it's guaranteed to get inconsistent pretty fast.

In terms of escape hatch I was thinking to have:

- a manual pypi to conda name mapping, to complement / override the one provided by pixi
- a list of pyproject dependencies to ignore

Reply:
Agreed with adding the `[tool.pixi.project]` section during init. Noting I see maybe two main scenarios for a pyproject.toml related init

1. 'take over' an existing project, where content is added to existing files
2. create a new empty project with a pyproject.toml instead of a pixi.toml file

number 2 would be the most straightforward to implement